### PR TITLE
the netty unicast processor is not emitting errors for downstream subscribers

### PR DESCRIPTION
**Pull request by rawilder**
related to: https://github.com/micronaut-projects/micronaut-core/issues/6062

### CODE REVIEW

**Comment by graemerocher** [comment]
Needs a test

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java [hunk 1]
```diff
@@ -827,16 +827,17 @@ protected void doOnNext(Object message) {
...
                 protected void doOnError(Throwable t) {
                     s.cancel();
+                    subjects.values().forEach(subject -> subject.onError(t));
                     emitter.error(t);
                 }
 
                 @Override
                 protected void doOnComplete() {
-                    for (UnicastProcessor subject : subjects.values()) {
+                    subjects.values().forEach(subject -> {
```
Comment (jameskleeh) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java]: Don't change to functional style

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java [hunk 2]
```diff
@@ -827,16 +827,17 @@ protected void doOnNext(Object message) {
                 @Override
                 protected void doOnError(Throwable t) {
                     s.cancel();
+                    subjects.values().forEach(subject -> subject.onError(t));
```
Comment (jameskleeh) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java]: A simple for loop as is below is preferred here

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java (hunk 1 above)
Comment (rawilder) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java]: Sorry about that, force of habit

**Comment by rawilder** [comment]
This will sound crazy but I can't replicate the kotlin failure via a test (see here: https://github.com/rawilder/mn-multi-file-upload/pull/1/files).

It's only happening via server start -> manual request. Which means I'd have to inject a failure somehow into the onNext function of `RoutingInBoundHandler:buildSubscriber` which I really am not sure how to do. I was trying to find some component I could mock and make throw but nothing stood out as obvious or not breaking other bits. Any ideas?

**Comment by jameskleeh** [comment]
@rawilder Try moving the test which should fail into its own class. Because tests are executing before the new tests they are not the first request to the server which is the case where we saw the failure

**Comment by rawilder** [comment]
@jameskleeh I moved the two tests to their own class locally and both succeeded. Would you like me to still push that to separate them out?

**Pull request merged**